src/main/java/org/embulk/input/jira/client/JiraClient.java

```diff
@@ -0,0 +1,197 @@
+
+                @Override
+                public void onRetry(Exception exception, int retryCount, int retryLimit, int retryWait)
+                        throws RetryGiveupException
+                {
+                    if (exception instanceof JiraException) {
+                        String message = String
+                                .format("Retrying %d/%d after %d seconds. HTTP status code: %s",
+                                        retryCount, retryLimit,
+                                        retryWait / 1000,
+                                        ((JiraException) exception).getStatusCode());
+                        LOGGER.warn(message);
+                    }
+                    else {
+                        String message = String
+                                .format("Retrying %d/%d after %d seconds. Message: %s",
+                                        retryCount, retryLimit,
+                                        retryWait / 1000,
+                                        exception.getMessage());
+                        LOGGER.warn(message, exception);
+                    }
+                }
+
+                @Override
+                public void onGiveup(Exception firstException, Exception lastException) throws RetryGiveupException
+                {
+                    LOGGER.warn("Retry Limits Completed");
+                }
+            });
+        }
+        catch (RetryGiveupException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private String authorizeAndRequestWithGet(final PluginTask task, String url) throws JiraException
+    {
+        try {
+            HttpClient client = createHttpClient();
+            HttpGet request = createGetRequest(client, task, url);
+            HttpResponse response = client.execute(request);
+            // Check for HTTP response code : 200 : SUCCESS
+            int statusCode = response.getStatusLine().getStatusCode();
+            if (statusCode != HttpStatus.SC_OK) {
+                throw new JiraException(statusCode,
+                        "JIRA API Request Failed: " + EntityUtils.toString(response.getEntity()));
+            }
+            return EntityUtils.toString(response.getEntity());
+        }
+        catch (IOException e) {
+            throw new JiraException(-1, e.getMessage());
+        }
+    }
+
+    private HttpClient createHttpClient()
+    {
+        RequestConfig config = RequestConfig.custom()
+                .setConnectTimeout(CONNECTION_TIME_OUT)
+                .setConnectionRequestTimeout(CONNECTION_TIME_OUT)
+                .build();
+        HttpClient client = HttpClientBuilder.create().setDefaultRequestConfig(config).build();
+        return client;
+    }
+
+    private HttpGet createGetRequest(HttpClient client, PluginTask task, String url)
+    {
+        HttpGet request = new HttpGet(url);
+        switch (task.getAuthMethod()) {
+        default:
```

Comment:
I think we could prevent this inside of AuthenticateMethod enum. 
And I've already has it in https://github.com/treasure-data/embulk-input-jira/blob/embulk-input-jira_java/src/main/java/org/embulk/input/jira/AuthenticateMethod.java#L24

Reply:
Got it, but I still don't understand why if the logic of this part of `JiraClient` is if the `authMethod` is `BASIC` then append the "Authorization: Basic ..." header then why isn't it simply written so but "catch all" with `default`? It might make readers puzzled with "Is the intention is to fallback to Basic authorization if the known authorization is unhandled/unimplemented?"